src/plugins/controls/public/control_group/editor/data_control_editor_tools.ts

```diff
@@ -30,20 +30,19 @@ const loadFieldRegistryFromDataView = async (
   const controlFactories = getControlTypes().map(
     (controlType) => getControlFactory(controlType) as IEditableControlFactory
   );
-  const fieldRegistry: DataControlFieldRegistry = dataView.fields
-    .getAll()
-    .reduce((registry, field) => {
-      const test: DataControlField = { field, compatibleControlTypes: [] };
+  const fieldRegistry: DataControlFieldRegistry = {};
+  return new Promise<DataControlFieldRegistry>((resolve) => {
+    for (const field of dataView.fields.getAll()) {
```

Comment:
`for ... of` is not much better. How about an imperative for loop, the one with the iterator `for(i=0; i<length;i++)`

Reply:
I did some performance testing of all 3 options:
1. Using `.forEach` with a callback - took `11.64ms` on average over 5 tries
2. Using `for ... of` for loop - took `12.52ms` on average over 5 tries
3. Using imperative `for(let i=0....)` for loop - took `12.80ms` on average over 5 tries.
The differences here are so minimal, I'm wondering if it is worth going this deep?

Reply:
right, looking at https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/ `for..in` and a traditional `for` loop have about the same performance, with `for..in` having slightly better performance.

Reply:
Was just going to link the same article 😆 Awesome, thanks!